### ISSUE TITLE
Edge doesn't show in the graph

### ISSUE THREAD

**Issue by AntoineCampi** [orientechnologies/orientdb#9123]
## Expected behavior  
When query in the graph menu (with SELECT * FROM V, for example) the edge should be displayed.

## Actual behavior  
The edge doesn't show and can't be selected in the graph menu, althought it appears in the schema menu and can be queried in the browse menu.

## Steps to reproduce  
Begining with an empty database : 

CREATE CLASS user EXTENDS V ;
CREATE CLASS movie EXTENDS V ;
CREATE CLASS RATING EXTENDS E ;
CREATE PROPERTY user.userid INTEGER ;
CREATE PROPERTY movie.movieid INTEGER ;
CREATE PROPERTY movie.title STRING ;
CREATE PROPERTY RATING.rate FLOAT ;
UPDATE user SET userid=1 UPSERT WHERE userid=1 ;
UPDATE movie SET movieid=2, title='Jumanji' UPSERT WHERE movieid=2 ;

UPDATE RATING SET out=(SELECT * FROM user WHERE userid=1), in=(SELECT * FROM movie WHERE title='Jumanji') UPSERT WHERE out.userid=1 AND in.title='Jumanji'

This last statement creates an edge that can be requested in browse menu but not in the graph one. Moreover, the query : 
SELECT * FROM RATING
raise an error in the graph menu : java.lang.NullPointerException

**Comment by luigidellaquila** [comment]
Hi @AntoineCampi 

That NPE is definitely worth checking, I'll do it asap.

Anyway, what you are doing here is not creating an edge, you are just forcing some internal conventions to let OrientDB believe that there is an edge somewhere.
I'd strongly suggest you to use CREATE EDGE statements instead

Thanks

Luigi

**Comment by AntoineCampi** [comment]
Hi and thanks for your answer.

I am not using CREATE because I want to be sure that the edge I am inserting isn't already in my database. Is there a way to do so ?

I believe this should work as I am able to query it in the browse menu, therefore the edge is the database. The main issue, in my opinion, is that the graph menu is not able to request/display it. The fact that it send back an error and not an empty graph suggest that this way should work.

Antoine

**Comment by luigidellaquila** [comment]
hi @AntoineCampi 

you can use CREATE EDGE with UPSERT, see https://orientdb.org/docs/3.0.x/sql/SQL-Create-Edge.html
It should be exactly what you need

Thanks

Luigi

**Comment by AntoineCampi** [comment]
I keep investigating and I don't really know what happened but it is now working.

I try to batch queries using pyorient (the same statement with UPDATE ... UPSERT ...) and not only I can see the edge I created in batch, but now the new edge i am inserting in the database (still the same statement) are displayed in the graph menu.

So in a way I don't have no more issues in my case but I don't know what change, when I create a new database the issue happen again.

I tried the CREATE EDGE STATEMENT but it seems to duplicate the edges even with the UPSERT key word.

Thanks,
Antoine
